### PR TITLE
Add fallback for any apps still using handleClick

### PR DESCRIPTION
**Pull request by dancork**
#### What tests does this PR have?
Add test for onClick and regression test for handleClick.

#### How can this be tested?
Fireup any tripapp using master (local, staging or production)
Make a carpark search for LGW and on availability select a non-flexible product
No upsell modal will fire

On your local machine pull this branch
On your local tripapp npm link your local ui-toolkit ( probably `npm link ../ui-toolkit` )
Fire up tripapp and do the same search.
An upsell modal should now fire

#### What gif best describes how you feel about this work?

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [ ] :+1:

**Review 2**
- [x] :+1:

**Review 3**
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

### CODE REVIEW

**Comment by cameronviner** [comment]
Taking Review

File: src/components/button/button.jsx [hunk 1]
```diff
@@ -24,34 +25,20 @@ module.exports = React.createClass({
...
   },
 
-  render: function() {
-    return this.props.href ? this.renderAnchor() : this.renderButton();
-  },
-
-  renderAnchor: function() {
-    const props = _.omit(this.props, 'children');
+  _getProps() {
+    const props = _.omit(this.props, ['data', 'size', 'purpose']);
```
Comment (dancork) [inline src/components/button/button.jsx]: This is something I noticed yesterday so thought I'd kill 2 birds...

https://holidayextras.jira.com/browse/WEB-15447

* no actual birds were killed in the making of this PR

**Comment by cameronviner** [comment]
Looks good to me 👍

**Comment by rahulpatel** [comment]
Nice! 👍

**Pull request merged**